### PR TITLE
get/list cluster templates

### LINKED ISSUE #7167: Add get/list endpoints for cluster templates
**User Story**

Add get/list endpoints for cluster templates

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: get/list cluster templates

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7167

```release-note
cluster templates
```

### CODE REVIEW

File: pkg/handler/v2/cluster_template/cluster_template.go [hunk 1]
```diff
@@ -160,6 +161,127 @@ func DecodeCreateReq(c context.Context, r *http.Request) (interface{}, error) {
...
+
+		result := apiv2.ClusterTemplateList{}
+
+		templates, err := clusterTemplateProvider.List(userInfo, project.Name)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		for _, template := range templates {
+			result = append(result, apiv2.ClusterTemplate{
```
Comment (lsviben) [inline pkg/handler/v2/cluster_template/cluster_template.go:188]: what about Cluster and NodeDeployment?

File: pkg/handler/v2/cluster_template/cluster_template_test.go [hunk 2]
```diff
@@ -136,3 +136,201 @@ func TestCreateClusterRemplateEndpoint(t *testing.T) {
...
+		{
+			Name: "scenario 1: list cluster templates",
+			ExpectedClusterTemplates: []apiv2.ClusterTemplate{
+				{
+					Name:           "ct1",
+					ID:             "ctID1",
+					ProjectID:      test.GenDefaultProject().Name,
+					User:           test.GenDefaultAPIUser().Email,
+					Scope:          kubermaticv1.UserClusterTemplateScope,
+					Cluster:        nil,
```
Comment (lsviben) [inline pkg/handler/v2/cluster_template/cluster_template_test.go:159]: NIT: the explicit nil`s are not needed

File: pkg/handler/v2/cluster_template/cluster_template.go (hunk 1 above)
Comment (zreigz) [reply pkg/handler/v2/cluster_template/cluster_template.go:188]: This is only for listing templates. The user just needs to know the name and ID. Cluster and ND are too heavy. 
The details will be fetched in GET call

**Pull request merged**